### PR TITLE
Add zen_workaround to our repo

### LINKED ISSUE #3690: could the zen workaround module be incorporated to the repo
This module sometimes needs updates, and having it tracked in a versioned repo would help downstream notice these updates. Currently it lives in a gist: https://gist.github.com/glandium/01d54cefdb70561b5f6675e08f2990f2

https://gist.github.com/glandium/01d54cefdb70561b5f6675e08f2990f2?permalink_comment_id=4907239#gistcomment-4907239

### PR DESCRIPTION
**Pull request by rocallahan**
It previously lived at https://gist.github.com/glandium/01d54cefdb70561b5f6675e08f2990f2

Resolves #3690

@glandium can you verify that this is OK? I'm not sure what to do about licensing, authorship etc.

### CODE REVIEW

**Comment by glandium** [comment]
> I'm not sure what to do about licensing, authorship etc.

I guess you can add a GPLv2 block at the beginning of the file, add a copyright notice (year would be 2020), and put me as the author, as the comment about inspiration (https://gist.github.com/glandium/01d54cefdb70561b5f6675e08f2990f2?permalink_comment_id=3459684#gistcomment-3459684) only applies to the earliest revisions of the gist.

**Comment by rocallahan** [comment]
@glandium let me know if this is OK.

File: third-party/zen-pmu-workaround/zen_workaround.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#include <linux/module.h>
```
Comment (glandium) [inline third-party/zen-pmu-workaround/zen_workaround.c:1]: Looks like kernel code now uses this as a header:

// SPDX-License-Identifier: GPL-2.0-only

I'd be fine with using whatever license makes this easier, but the module does need the 
`MODULE_LICENSE("GPL");` for the pm_notifier and probe registrations.

File: third-party/zen-pmu-workaround/README [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Mike Hommey's Zen workaround kernel module.
```
Comment (glandium) [inline third-party/zen-pmu-workaround/README:1]: Maybe mention that it does the same thing as scripts/zen_workaround.py. and/or zen_workaround.service in a way that persists after resume from sleep. (BTW, now that I discovered there's a systemd service file, there's probably a way to restore zen_workaround.py on resume with it too)

I'd also start with Zen workaround kernel module, and put something like "Copyright 2020, Mike Hommey" (as to indicate the age of this thing)

Not sure this warrants a separate file though.

File: third-party/zen-pmu-workaround/zen_workaround.c (hunk 1 above)
Comment (rocallahan) [reply third-party/zen-pmu-workaround/zen_workaround.c:1]: Are you suggesting a change here?

**Comment by rocallahan** [comment]
OK, how about this?

File: third-party/zen-pmu-workaround/zen_workaround.c (hunk 1 above)
Comment (glandium) [reply third-party/zen-pmu-workaround/zen_workaround.c:5]: I was just suggesting to put `// SPDX-License-Identifier: GPL-2.0-only` as a header in this file.

**Review by glandium — APPROVED** [review]

File: third-party/zen-pmu-workaround/README (hunk 2 above)
Comment (lissyx) [reply third-party/zen-pmu-workaround/README]: I think the kernel module is also the only way to do it when you have SecureBoot enabled, python script was failing for me with that setup.

**Comment by rocallahan** [comment]
This landed.

**Pull request closed without merging**